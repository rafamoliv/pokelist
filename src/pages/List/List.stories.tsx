import { Story } from '@storybook/react'

import List from './List'

export default {
  component: List,
  title: 'Pages/List',
  args: {}
}

const Template: Story = (args) => <List {...args} />

export const Default = Template.bind({})
