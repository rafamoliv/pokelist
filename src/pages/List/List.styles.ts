import styled from '@emotion/styled'
import { Box, Container, Icon } from '@chakra-ui/react'

export const StyledContainer = styled(Container)``

export const TableHeading = styled(Box)`
  display: flex;
  justify-content: space-between;
`

export const StyledIcon = styled(Icon)`
  cursor: pointer;
`
