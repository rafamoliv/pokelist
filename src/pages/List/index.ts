export { default } from './List'
export { listEn, listPtBr } from './locales'
