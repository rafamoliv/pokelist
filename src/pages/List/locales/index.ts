export { default as listPtBr } from './ptBr'
export { default as listEn } from './en'
