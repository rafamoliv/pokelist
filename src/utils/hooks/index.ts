import { TypedUseSelectorHook, useDispatch, useSelector } from 'react-redux'
import { RootState, AppDispatch } from '@/store'
import { useLocalStorage } from './uselLocalStorage'

export const useAppDispatch: () => AppDispatch = useDispatch
export const useAppSelector: TypedUseSelectorHook<RootState> = useSelector
export { useLocalStorage }
