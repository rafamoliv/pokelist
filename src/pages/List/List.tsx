import { useTranslation } from 'react-i18next'
import { Key, useState } from 'react'
import { Link } from 'react-router-dom'
import { useFetchPokemonsQuery } from '@/services/api'
import { SystemPage } from '@/templates'
import { StyledIcon as Icon, StyledContainer as Container, TableHeading } from './List.styles'
import { ArrowBackIcon, ArrowRightIcon, StarIcon } from '@chakra-ui/icons'
import { Box, Button, Heading, Select, Stack, Table, TableCaption, TableContainer, Tbody, Td, Tfoot, Th, Thead, Tr } from '@chakra-ui/react'
import { url } from '@/routes/urls'


const List = () => {
  const { t } = useTranslation('pgList')
  const [pokeLimit, setPokeLimit] = useState('10')
  const { data: pokeList = [] } = useFetchPokemonsQuery(pokeLimit)

  console.log(pokeList)

  return <SystemPage.Root>
    <section>
      <Container maxW='container.xl'>
        <Box paddingTop={4}>
          <Link to={url.HOME}>
            <Button leftIcon={<ArrowBackIcon />} colorScheme='whiteAlpha' size={'sm'} variant='outline'>
              {t('btn', { context: 'back' })}
            </Button>
          </Link>
        </Box>
        <Box paddingBottom={8} paddingTop={8}>
          <Stack spacing={4}>
            <TableHeading>
              <Heading as='h3' color={'white'} size='xl' noOfLines={1}>
                {t('table.title')}
              </Heading>
              <Box>
                <Select onChange={(e) => setPokeLimit(e.target.value)} size='md' variant={'Filled'}>
                  <option value='10' selected>10</option>
                  <option value='20'>20</option>
                  <option value='30'>30</option>
                </Select>
              </Box>
            </TableHeading>
            <TableContainer color={'white'}>
              <Table size={'sm'} variant='simple' colorScheme='whiteAlpha'>
                <TableCaption>{t('table.caption')}</TableCaption>
                <Thead>
                  <Tr>
                    <Th>{t('table.head.0')}</Th>
                    <Th textAlign={'center'} width={'15%'}>{t('table.head.1')}</Th>
                    <Th textAlign={'center'} width={'15%'}>{t('table.head.2')}</Th>
                  </Tr>
                </Thead>
                <Tbody>
                  {pokeList?.results?.map((poke: any, index: Key) => (
                    <Tr key={index}>
                      <Td>{poke.name}</Td>
                      <Td textAlign={'center'}><Icon as={StarIcon} /></Td>
                      <Td textAlign={'center'}>
                        <Button border={'none'} colorScheme='whiteAlpha' isLoading={false} rightIcon={<ArrowRightIcon />} variant={'outline'} />
                      </Td>
                    </Tr>
                  ))}
                </Tbody>
                <Tfoot>
                  <Tr>
                    <Th>{t('table.head.0')}</Th>
                    <Th textAlign={'center'}>{t('table.head.1')}</Th>
                    <Th textAlign={'center'}>{t('table.head.2')}</Th>
                  </Tr>
                </Tfoot>
              </Table>
            </TableContainer>
          </Stack>
        </Box>
      </Container>
    </section>
  </SystemPage.Root>
}
export default List
