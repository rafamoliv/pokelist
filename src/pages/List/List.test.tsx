import { render, screen, userEvent } from '@/utils/tests'
import List from './List'

describe('<List />', () => {
  describe('Component Snapshot', () => {
    it('Render element', () => {
      const { container } = render(<List />)
      expect(container.firstChild).toMatchSnapshot()
    })
  })

  describe('Component Roles', () => {
    it('Should have title', () => {
      render(<List />)
      expect(screen.getByRole('heading', { level: 3 })).toBeInTheDocument()
    })

    it('Should have button', () => {
      render(<List />)
      expect(screen.getAllByRole(/button/i)[0]).toBeInTheDocument()
    })
  })
})
