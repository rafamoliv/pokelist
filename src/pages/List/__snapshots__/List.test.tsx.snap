// Vitest Snapshot v1

exports[`<List /> > Component Snapshot > Render element 1`] = `
<main
  class="css-0"
>
  <section>
    <div
      class="chakra-container css-0 css-tflc3m"
    >
      <div
        class="css-37urdo"
      >
        <a
          href="/"
        >
          <button
            class="chakra-button css-4xx2wk"
            type="button"
          >
            <span
              class="chakra-button__icon css-1wh2kri"
            >
              <svg
                aria-hidden="true"
                class="chakra-icon css-onkibi"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
                  fill="currentColor"
                />
              </svg>
            </span>
            btn
          </button>
        </a>
      </div>
      <div
        class="css-1whxuhf"
      >
        <div
          class="chakra-stack css-17kdbol"
        >
          <div
            class="css-uf1ume css-0"
          >
            <h3
              class="chakra-heading css-1oa9rap"
            >
              table.title
            </h3>
            <div
              class="css-0"
            >
              <div
                class="chakra-select__wrapper css-42b2qy"
              >
                <select
                  class="chakra-select css-1ik61og"
                >
                  <option
                    value="10"
                  >
                    10
                  </option>
                  <option
                    value="20"
                  >
                    20
                  </option>
                  <option
                    value="30"
                  >
                    30
                  </option>
                </select>
                <div
                  class="chakra-select__icon-wrapper css-162mkon"
                >
                  <svg
                    aria-hidden="true"
                    class="chakra-select__icon"
                    focusable="false"
                    role="presentation"
                    style="width: 1em; height: 1em; color: currentColor;"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                      fill="currentColor"
                    />
                  </svg>
                </div>
              </div>
            </div>
          </div>
          <div
            class="chakra-table__container css-1rujb15"
          >
            <table
              class="chakra-table css-0"
            >
              <caption
                class="css-18i7bby"
              >
                table.caption
              </caption>
              <thead
                class="css-0"
              >
                <tr
                  class="css-0"
                >
                  <th
                    class="css-0"
                  >
                    table.head.0
                  </th>
                  <th
                    class="css-q99f1g"
                  >
                    table.head.1
                  </th>
                  <th
                    class="css-q99f1g"
                  >
                    table.head.2
                  </th>
                </tr>
              </thead>
              <tbody
                class="css-0"
              />
              <tfoot
                class="css-0"
              >
                <tr
                  class="css-0"
                >
                  <th
                    class="css-0"
                  >
                    table.head.0
                  </th>
                  <th
                    class="css-xi606m"
                  >
                    table.head.1
                  </th>
                  <th
                    class="css-xi606m"
                  >
                    table.head.2
                  </th>
                </tr>
              </tfoot>
            </table>
          </div>
        </div>
      </div>
    </div>
  </section>
</main>
`;
